docs/asdf/developer_overview.rst

```diff
@@ -34,11 +34,50 @@ Some terminology and definitions
 name that follows a particular syntax, but itself may not indicate where the
 resource is. Generally URLs are expected to be used on the web for the HTTP
 protocol, though for asdf, this isn't necessarily the case as mentioned next.
+Recent changes to the library permit use of URIs with the asdf:// scheme, which
+is intended to reduce confusion over the distinction between identifiers
+and locations.
 
 **Resolver:** Tools to map URIs and tags into actual locations of schema files,
 which may be local directories (the usual approach) or an actual URL for
 retrieval over the network. This is more complicated that it may seem for
-reasons explained later.
+reasons explained later.  The resolver system has been deprecated in favor
+of resource mappings; new code should use the latter instead.
+
+**Global config:** A global library configuration feature that was added in
+asdf 2.8.  Allows plugins to be added or removed at runtime and ``AsdfFile``
+defaults to be modified by the user.  Accessed via the ``get_config`` method
+on the top-level ``asdf`` module.
+
+**Entry point:** A Python packaging feature that allows a given package
+to use plugins registered by other packages.  See [https://packaging.python.org/specifications/entry-points/]
+for more information.
+
+**Resource mapping:** An asdf plugin that provides access to "resources" which
+are binary blobs associated with a URI.  These resources are mostly schemas,
+but any resource may provided by a mapping.  Resource mappings are provided
+via entry points or added at runtime using a method on the global config object.
```

Comment:
And as a replacement for the previous resolver mechanism.

Reply:
Added a sentence to that effect.